protocols/kad/src/query/peers/closest.rs

```diff
@@ -296,32 +320,72 @@ impl ClosestPeersIter {
                         result_counter = None;
                     }
                 }
-
                 PeerState::Succeeded =>
                     if let Some(ref mut cnt) = result_counter {
                         *cnt += 1;
                         // If `num_results` successful results have been delivered for the
                         // closest peers, the iterator is done.
-                        if *cnt >= self.config.num_results {
+                        if *cnt > self.config.num_results {
```

Comment:
Given that I first iterate all peers looking for timeouts and successes, the function does not behave as before anymore even when `disjoint_paths` is set to `1` (disabled). Only tests failing without the line above are `put_record` and `add_provider`. I still need to investigate this further.

Reply:
TBH for simplicity I would not even keep the old behaviour as it is obviously insecure and I can't see a use case for ever not enabling disjoint paths, even for testing/debugging purposes.